frontend/src/pages/MedDog.tsx

```diff
@@ -1,31 +1,82 @@
 import './MedDog.scss';
-import {LocalizationProvider, StaticDatePicker} from "@mui/lab";
-import AdapterDateFns from '@mui/lab/AdapterDateFns';
-import {TextField} from "@mui/material";
-import React from "react";
+import React, {ChangeEvent, useContext, useState} from "react";
+import AppointmentGallery from "../components/AppointmentGallery";
+import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
+import {Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField} from "@mui/material";
+import {DateTimePicker, LocalizationProvider} from "@mui/lab";
+import AdapterDateFns from "@mui/lab/AdapterDateFns";
+import {addAppointment} from "../service/RequestService";
+import {AuthContext} from "../context/AuthProvider";
 
 export default function MedDog() {
 
-    const [value, setValue] = React.useState<Date | null>(new Date());
+    const [open, setOpen] = React.useState(false);
+    const [appointment, setAppointment] = useState<string>("")
+    const [dateAndTime, setDateAndTime] = React.useState<Date | null>(new Date());
+
+    const {token} = useContext(AuthContext)
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setAppointment(event.target.value)
+    }
+
+    const addNewAppointment = () => {
+        //addAppointment(appointment)
+    };
```

Comment:
Either uncomment or delete the function as it does nothing

Reply:
Line 32 is commented because the 'add Appointment' function is not yet fully written and there were errors when compiling.

Reply:
okay, I am just not a huge fan of pushing stuff to main that is not working (yet).